src/store/getters.js

```diff
@@ -69,4 +69,11 @@ export const getters = {
 		const envelopes = thread.map(id => state.envelopes[id])
 		return sortBy(prop('dateInt'), envelopes)
 	},
+	getEnvelopeTags: (state) => (id) => {
+		const labels = state.envelopes[id]?.tags ?? []
+		return labels.map((imapLabel) => state.tags[imapLabel])
```

Comment:
I'm not sure if we want to index the tags by their imap label or by ID

Reply:
We definitely want to index them by id. I assumed the labels to be ids because tagging and untagging messages currently works by sending imap labels. I will refactor my code.